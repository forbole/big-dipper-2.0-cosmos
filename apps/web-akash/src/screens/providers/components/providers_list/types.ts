export type TitleListType = {
  ownerAddress: string;
  hostUri: string;
  region: string;
  organization: string;
  email: string;
  website: string;
};

export type TitleListState = {
  hasNextPage: boolean;
  isNextPageLoading: boolean;
  offsetCount: number;
  data: TitleListType[];
};
