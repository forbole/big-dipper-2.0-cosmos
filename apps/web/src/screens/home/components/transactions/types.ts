export type TransactionType = {
  height: number;
  hash: string;
  type: string[];
  success: boolean;
  timestamp: string;
  messages: number;
};

export type TransactionsState = {
  items: TransactionType[];
};
