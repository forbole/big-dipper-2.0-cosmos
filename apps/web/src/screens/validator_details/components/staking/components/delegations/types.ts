import type { DelegationType } from '../../types';

export type ItemType = Override<DelegationType, { address: AvatarName }>;
