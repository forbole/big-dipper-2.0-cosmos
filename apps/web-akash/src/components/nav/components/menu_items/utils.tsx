import {
  BLOCKS,
  HOME,
  PARAMS,
  PROPOSALS,
  PROVIDERS,
  TRANSACTIONS,
  VALIDATORS,
} from '@/utils/go_to_page';
import BlockIcon from 'shared-utils/assets/icon-block.svg';
import HomeIcon from 'shared-utils/assets/icon-home.svg';
import ParamIcon from 'shared-utils/assets/icon-param.svg';
import ProposalsIcon from 'shared-utils/assets/icon-proposals.svg';
import ProviderIcon from 'shared-utils/assets/icon-provider.svg';
import TransactionIcon from 'shared-utils/assets/icon-transaction.svg';
import UserIcon from 'shared-utils/assets/icon-user.svg';

export const getMenuItems = () => {
  const iconProps = {
    width: 24,
    height: 24,
  };
  return [
    {
      key: 'overview',
      url: HOME,
      icon: <HomeIcon {...iconProps} />,
    },
    {
      key: 'blocks',
      url: BLOCKS,
      icon: <BlockIcon {...iconProps} />,
    },
    {
      key: 'validators',
      url: VALIDATORS,
      icon: <UserIcon {...iconProps} />,
    },
    {
      key: 'transactions',
      url: TRANSACTIONS,
      icon: <TransactionIcon {...iconProps} />,
    },
    {
      key: 'proposals',
      url: PROPOSALS,
      icon: <ProposalsIcon {...iconProps} />,
    },
    {
      key: 'params',
      url: PARAMS,
      icon: <ParamIcon {...iconProps} />,
    },
    {
      key: 'providers',
      url: PROVIDERS,
      icon: <ProviderIcon {...iconProps} />,
    },
  ];
};
