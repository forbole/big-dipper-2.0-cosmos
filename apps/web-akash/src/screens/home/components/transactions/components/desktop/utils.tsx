export const columns: {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'block',
    width: 20,
  },
  {
    key: 'hash',
    width: 20,
  },
  {
    key: 'type',
    width: 20,
  },
  {
    key: 'result',
    align: 'right',
    width: 20,
  },
  {
    key: 'time',
    align: 'right',
    width: 20,
  },
];
