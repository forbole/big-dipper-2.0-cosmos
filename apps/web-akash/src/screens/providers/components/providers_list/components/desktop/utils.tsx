export const columns: {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
}[] = [
  {
    key: 'ownerAddress',
    width: 20,
  },
  {
    key: 'hostUri',
    width: 25,
  },
  {
    key: 'region',
    align: 'right',
    width: 15,
  },
  {
    key: 'organization',
    align: 'right',
    width: 20,
  },
  {
    key: 'email',
    align: 'center',
    width: 5,
  },
  {
    key: 'website',
    align: 'right',
    width: 15,
  },
];
