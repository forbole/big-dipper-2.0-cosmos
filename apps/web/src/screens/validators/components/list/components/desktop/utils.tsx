import React from 'react';
import Typography from '@material-ui/core/Typography';
import InfoPopover from 'ui/components/info_popover';
import ConditionExplanation from 'ui/components/condition_explanation';
import { Translate } from 'next-translate';

export const fetchColumns = (
  t: Translate
): {
  key: string;
  align?: 'left' | 'center' | 'right' | 'justify' | 'inherit';
  width: number;
  component?: React.ReactNode;
  sortKey?: string;
  sort?: boolean;
}[] => [
    {
      key: 'idx',
      width: 5,
    },
    {
      key: 'validator',
      sortKey: 'validator.name',
      width: 25,
      sort: true,
    },
    {
      key: 'votingPower',
      sortKey: 'votingPower',
      width: 30,
      sort: true,
    },
    {
      key: 'commission',
      sortKey: 'commission',
      align: 'right',
      width: 15,
      sort: true,
    },
    // {
    //   key: 'self',
    //   sortKey: 'selfPercent',
    //   align: 'right',
    //   width: 11,
    //   sort: true,
    // },
    // {
    //   key: 'delegators',
    //   sortKey: 'delegators',
    //   align: 'right',
    //   width: 11,
    //   sort: true,
    // },
    {
      key: 'condition',
      align: 'center',
      width: 15,
      component: (
        <Typography variant="h4" className="label popover">
          {t('condition')}
          <InfoPopover content={ConditionExplanation} />
        </Typography>
      ),
    },
    {
      key: 'status',
      width: 10,
    },
  ];
