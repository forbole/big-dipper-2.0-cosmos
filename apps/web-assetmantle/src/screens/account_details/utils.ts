import axios from 'axios';
import * as R from 'ramda';
import { toValidatorAddress } from 'ui/utils/prefix_convert';
import {
  AccountCommissionDocument,
  AccountWithdrawalAddressDocument,
  AccountBalancesDocument,
  AccountDelegationBalanceDocument,
  AccountUnbondingBalanceDocument,
  AccountDelegationRewardsDocument,
} from '@src/graphql/general/account_details_documents';

import chainConfig from 'ui/chainConfig';

export const fetchCommission = async (address: string) => {
  const defaultReturnValue = {
    commission: {
      coins: null,
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          validatorAddress: toValidatorAddress(address),
        },
        query: AccountCommissionDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAccountWithdrawalAddress = async (address: string) => {
  const defaultReturnValue = {
    withdrawalAddress: {
      address,
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountWithdrawalAddressDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchAvailableBalances = async (address: string) => {
  const defaultReturnValue = {
    accountBalances: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountBalancesDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchDelegationBalance = async (address: string) => {
  const defaultReturnValue = {
    delegationBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationBalanceDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchUnbondingBalance = async (address: string) => {
  const defaultReturnValue = {
    unbondingBalance: {
      coins: [],
    },
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountUnbondingBalanceDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};

export const fetchRewards = async (address: string) => {
  const defaultReturnValue = {
    delegationRewards: [],
  };
  try {
    const { data } = await axios.post(
      process.env.NEXT_PUBLIC_GRAPHQL_URL ||
        chainConfig.endpoints.graphql ||
        'http://localhost:3000/v1/graphql',
      {
        variables: {
          address,
        },
        query: AccountDelegationRewardsDocument,
      }
    );
    return R.pathOr(defaultReturnValue, ['data'], data);
  } catch (error) {
    return defaultReturnValue;
  }
};
