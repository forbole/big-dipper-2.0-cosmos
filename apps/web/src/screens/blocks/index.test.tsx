import { ApolloClient, ApolloProvider, from, InMemoryCache } from '@apollo/client';
import { MockedProvider } from '@apollo/client/testing';
import { BlocksDocument, BlocksListenerDocument } from '@graphql/types/general_types';
import React from 'react';
import renderer from 'react-test-renderer';
import { MockTheme, wait } from 'ui/tests/utils';
import Blocks from '.';

let component: renderer.ReactTestRenderer;

// ==================================
// mocks
// ==================================
jest.mock('ui/components/layout', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Layout" {...props} />
));
jest.mock('ui/components/box', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="Box" {...props} />
));
jest.mock('ui/components/load_and_exist', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="LoadAndExist" {...props} />
));
jest.mock('ui/components/no_data', () => (props: JSX.IntrinsicElements['div']) => (
  <div id="NoData" {...props} />
));

jest.mock('./hooks', () => ({
  useBlocks: () => ({
      state: {
        loading: false,
        exists: true,
        hasNextPage: true,
        isNextPageLoading: false,
        items: [
          {
            height: 1123213,
            txs: 0,
            timestamp: '2021-04-27T16:27:34.331769',
            proposer: 'desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3',
            hash: 'txhash',
          },
          {
            height: 1123214,
            txs: 1,
            timestamp: '2021-04-33T16:27:34.331769',
            proposer: 'desmosvaloper18kvwy5hzcu3ss08lcfcnx0eajuecg69uvk76c3',
            hash: 'txhash',
          },
        ],
      },
      loadNextPage: () => jest.fn(),
      itemCount: 2,
      loadMoreItems: () => jest.fn(),
      isItemLoaded: () => true,
    }),
}));

jest.mock('./hooks', () => ({
  useBlocks: () => ({
      state: {
        loading: false,
        exists: true,
        hasNextPage: true,
        isNextPageLoading: false,
        items: [],
      },
      loadNextPage: () => jest.fn(),
      itemCount: 2,
      loadMoreItems: () => jest.fn(),
      isItemLoaded: () => true,
    }),
}));

const mockBlocksListenerDocument = jest.fn().mockReturnValue({
  data: {
    blocks: [
      {
        height: 379643,
        txs: 2,
        hash: 'D0243447726B8BD7AE94BF4F98E536A647959194E870AB8566CB833A3CC847F6',
        timestamp: '2021-05-24T05:28:05.839448',
        validator: {
          validatorInfo: {
            operatorAddress: 'desmosvaloper1h5f3dywec65v9qulxkmcv3e6yujyh3zm0ghhl3',
          },
        },
      },
    ],
  },
});

const mockBlocksDocument = jest.fn().mockReturnValue({
  data: {
    blocks: [
      {
        height: 379634,
        txs: 2,
        hash: '017E26F5C11E4C140AA1836E40315C00253EDBC9D69A475A4ABEDDD1FF0FE967',
        timestamp: '2021-05-24T05:27:11.092332',
        validator: {
          validatorInfo: {
            operatorAddress: 'desmosvaloper1zm3l7p8n5dxqeadsfxy3rd0j3c2knnx3chg77a',
            self_delegate_address: 'desmos1zm3l7p8n5dxqeadsfxy3rd0j3c2knnx3x6q250',
          },
          validatorDescriptions: [
            {
              moniker: 'Simply Staking',
              identity: null,
            },
          ],
        },
      },
    ],
    total: {
      aggregate: {
        count: 379619,
      },
    },
  },
});

// ==================================
// unit tests
// ==================================
describe('screen: Blocks', () => {
  it('matches snapshot', async () => {
    renderer.act(() => {
      component = renderer.create(
        <ApolloProvider client={new ApolloClient({ link: from([]), cache: new InMemoryCache() })}>
          <MockedProvider
            mocks={[
              { request: { query: BlocksDocument }, result: mockBlocksDocument },
              { request: { query: BlocksListenerDocument }, result: mockBlocksListenerDocument },
            ]}
          >
            <MockTheme>
              <Blocks />
            </MockTheme>
          </MockedProvider>
        </ApolloProvider>
      );
    });
    await wait(renderer.act);
    const tree = component?.toJSON();
    expect(tree).toMatchSnapshot();
  });

  afterEach(() => {
    jest.clearAllMocks();
  });
});
