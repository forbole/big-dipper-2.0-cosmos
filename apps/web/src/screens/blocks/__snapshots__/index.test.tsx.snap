// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`screen: Blocks matches snapshot 1`] = `
<div>
  <div
    className="makeStyles-root"
    id="Layout"
    navTitle="blocks"
  >
    <div
      exists={true}
      id="LoadAndExist"
      loading={false}
    >
      <div
        className="makeStyles-box"
        id="Box"
      >
        <div
          id="NoData"
        />
      </div>
    </div>
  </div>
</div>
`;
